ivy/array/array.py

```diff
@@ -135,10 +135,47 @@ def _init(self, data):
         else:
             self._post_repr = ")"
         self.backend = ivy.current_backend_str()
+        if dynamic_backend is not None:
+            self._dynamic_backend = dynamic_backend
+        else:
+            self._dynamic_backend = ivy.get_dynamic_backend()
 
     # Properties #
     # ---------- #
 
+    @property
+    def dynamic_backend(self):
+        return self._dynamic_backend
+
+    @dynamic_backend.setter
+    def dynamic_backend(self, value):
+        from ivy.functional.ivy.gradients import _variable
```

Comment:
Through this logic, we'd be importing the `_variable` and `_determine_backend_from_args` functions every time we set a `dynamic_backend`, whereas it would be much better if we just import them once at the top, unless there's an issue with circular imports if you do that (which I think is a possibility)

Reply:
Yup there's an issue with circular imports which is why I cant import it at the top

Reply:
Got it! Thanks for clearing it out!